NF: Adding actor for Tetrahedron, TEST: Adding uni tests for Tetrahedron

ManishReddyR opened the pull request:
Hello, 
Tried create the actor for Tetrahedron and also added few of the uni tests cases.
<img width="596" alt="Tetrahedron_mc" src="https://github.com/user-attachments/assets/9a753e78-df63-4498-8136-b39a896403cf" />
<img width="596" alt="Tetrahedron_sc" src="https://github.com/user-attachments/assets/9a2cf9f9-cee5-40c3-870c-7f756eb9b99b" />
Thank You 
Manish Rakasi

skoudoro reviewed (changes requested):
Hi @ManishReddyR,

- Can you rebase this PR on the top of the v2 branch (it avoids to see 15 commits)
- Can you do all of them in one PR? update this PR with all shape actor based on primitive. easy to review. I think it would be better than one by one. in this specific case. 

thank you

skoudoro commented:
I see you requested a review @ManishReddyR. Thank you for the update.

What about superquadric, rectangle, arrow, cone, triangularprism, rhombicuboctahedron, pentagonalprism, octagonalprism, tetrahedron, icosahedron, star?

skoudoro commented:
keep doing 1 commit per primitives shape, this is great and easier to review.

Thanks

skoudoro reviewed (commented):
See my comment above, many shapes are missing.

Otherwise, it looks good as it is now

ManishReddyR commented:
> I see you requested a review @ManishReddyR. Thank you for the update.
> 
> What about superquadric, rectangle, arrow, cone, triangularprism, rhombicuboctahedron, pentagonalprism, octagonalprism, tetrahedron, icosahedron, star?



working on it, will update all of them by end of this week

ManishReddyR commented:
Added the actor for rhombicuboctahedron, and also added the uni tests.
<img width="596" alt="rhombicuboctahedron mc" src="https://github.com/user-attachments/assets/78d757a7-460b-42dd-8436-de25ae68e3ce" />
<img width="596" alt="rhombicuboctahedron sc" src="https://github.com/user-attachments/assets/224713a4-65aa-48e1-bc9e-5bb6f8388391" />

Thank You

ManishReddyR commented:
Added the actor for the triangularprism, And also added the uni tests for triangularprism actor.

<img width="596" alt="triangularprism_mc" src="https://github.com/user-attachments/assets/ee4a5dd1-21c2-4735-9ff4-3ca0e8c105c4" />

<img width="596" alt="triangularprism_sc" src="https://github.com/user-attachments/assets/c0712f1f-f763-4356-8a85-d6b75c9b20cd" />

Thank you

ManishReddyR commented:
Added actor for pentagonalprism, and added uni tests for pentagonalprism.

<img width="596" alt="pentagonalprism_mc" src="https://github.com/user-attachments/assets/0fbcb1b8-f695-499a-9c9d-2bed854f1bb1" />

<img width="596" alt="pentagonalprism_sc" src="https://github.com/user-attachments/assets/72a674e2-fd86-4565-8707-6d935d6c1a23" />

Thank you

ManishReddyR commented:
Added the actor for octogonal prism, and also added uni tests for octogonal prism actor.

<img width="596" alt="octagonalprism_mc" src="https://github.com/user-attachments/assets/fcd054fd-640a-4d07-879a-c4e21368fe9b" />

<img width="596" alt="octagonalprism_sc" src="https://github.com/user-attachments/assets/8b0beab8-e872-4846-bbe4-386a9d6e2979" />

Thank you

ManishReddyR commented:
Added actor for arrow and also added uni tests for arrow.

<img width="596" alt="Arrow_SC" src="https://github.com/user-attachments/assets/a49f2ee2-08df-46af-9697-9e28c4148c1e" />

<img width="596" alt="Arrow_MC" src="https://github.com/user-attachments/assets/9e83d69b-4d01-434d-af13-cb31c343fa84" />

Thank you

ManishReddyR commented:
Added actor for superquadric and also added uni tests for superquadric.

<img width="596" alt="superquadric_mc" src="https://github.com/user-attachments/assets/b5b650c6-9134-497c-b686-d4e645df26e2" />

<img width="596" alt="superquadric_sc" src="https://github.com/user-attachments/assets/cd292f1d-4117-41d4-b159-63a54541516f" />

Thank you

ManishReddyR commented:
Hello @skoudoro ,

Can you review the actors, which I have added.

Thank you

skoudoro commented:
Hi @ManishReddyR, 

I will look at this on Friday. Thank you for working on this.

Can you update your tests ? they seems repetitive. Look here for an example on how you could refactor all of this:
https://github.com/fury-gl/fury/blob/337f4b3b6dcc27fc5133406c94f78741fea673d5/fury/tests/test_primitive.py#L9-L25

maharshi-gor commented:
Hi @ManishReddyR Thank you for the work.

@skoudoro should we use only npt for the assertions? I see we have used mixture of `assert` and `npt assert`. If not, any specific reason for the same?

skoudoro commented:
> @skoudoro should we use only npt for the assertions? I see we have used mixture of assert and npt assert. If not, any specific reason for the same?

it is fine, as soon as they are explicit enough.  `assert` do no handle array so `npt assert` is needed

ManishReddyR commented:
Added actor for cone, and also uni tests for the cone actors.

<img width="596" alt="cone_mc" src="https://github.com/user-attachments/assets/a5015695-f828-47af-b00d-bbf4bf80d345" />

<img width="596" alt="cone_sc" src="https://github.com/user-attachments/assets/bf53accc-7460-4626-9267-cf7bd9637d13" />


Thank you

ManishReddyR commented:
Added actor for star, and also added uni tests for star.

<img width="596" alt="Star_mc" src="https://github.com/user-attachments/assets/53f97a3d-0b00-45f6-9f3e-55e2acc54540" />

<img width="596" alt="star_sc" src="https://github.com/user-attachments/assets/8dbf6e21-f557-4ddf-a254-d690078516de" />

I have added almost all the actors based on the primitive. Now working on refactoring of the test cases.

Thank you

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    colors : ndarray (N,3) or (N, 4) or tuple (3,) or tuple (4,), optional
+        RGB or RGBA (for opacity) R, G, B and A should be at the range [0, 1].
+    height : float, optional
+        The total height of the arrow, including the shaft and tip. Default is 1.0.
```
"Default is 1.0."  to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+        The total height of the arrow, including the shaft and tip. Default is 1.0.
+    resolution : int, optional
+        The number of divisions along the arrow's circular cross-sections.
+        Higher values produce smoother arrows. Default is 10.
```
default to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+        The number of divisions along the arrow's circular cross-sections.
+        Higher values produce smoother arrows. Default is 10.
+    tip_length : float, optional
+        The length of the arrowhead tip relative to the total height. Default is 0.35.
```
Default to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    tip_length : float, optional
+        The length of the arrowhead tip relative to the total height. Default is 0.35.
+    tip_radius : float, optional
+        The radius of the arrowhead tip. Default is 0.1.
```
Default to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    tip_radius : float, optional
+        The radius of the arrowhead tip. Default is 0.1.
+    shaft_radius : float, optional
+        The radius of the arrow shaft. Default is 0.03.
```
Default to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    centers,
+    *,
+    roundness=(1, 1),
+    directions=(0, 0, 0),
```
not the same order as the docstring. directions should be first

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    colors : ndarray (N,3) or (N, 4) or tuple (3,) or tuple (4,), optional
+        RGB or RGBA (for opacity) R, G, B and A should be at the range [0, 1].
+    height: float, optional
+        The height of the cone. Default is 1.
```
default to remove

skoudoro commented on fury/actor.py:
```diff
@@ -441,3 +441,690 @@ def frustum(
+    centers : ndarray, shape (N, 3)
+        star positions.
+    dim : int, optional.
+        The dimensionality of the star (2D or 3D). Default is 2.
```
default to remove

skoudoro reviewed (changes requested):
Overall, it is good,

see the comment below to update. Waiting for the tests refactorization also. Thank @ManishReddyR

ManishReddyR commented:
Hello @skoudoro ,
Thanks for your review, and also refactored the tests.

skoudoro reviewed (approved):
Thank you for this work. 

Merging
